### PR TITLE
chore(ci): reduce runner sizing to 4 core and free tier

### PR DESCRIPTION
**Pull request by neuronull**
- In order to reduce CI costs, experiments were run on each of the altered workflows to understand the duration (and thus impact to DX) and changes in cost.
- The two cases of concern are what checks are run on each PR commit and in the merge queue:
    - in the merge queue- there was a trivial change in runtime. This is because the bottleneck workflow is the mac unit test, which takes about 1 hour to run. All runner-size-reduced workflows in this PR that are in the merge queue, still run within that time frame.
    - on each PR commit- this PR does not affect this case. However, a sister PR (#17724) does, because that PR fundamentally changes the integration test workflow to introduce the runner there, and so in that PR, we reduced to 4 core. That change adds 3 minutes of duration to the integration tests, which is trivial (and integration tests only run on some PR pushes, if that PR touches integrations).
- Thus the empirically anticipated outcome is a ~20% reduction in cost with a none-to-trivial increase in duration.

### CODE REVIEW

File: .github/workflows/regression.yml [hunk 1]
```diff
@@ -318,7 +318,7 @@ jobs:
 
   build-comparison:
     name: Build comparison Vector container
-    runs-on: [linux, soak-builder]
+    runs-on: [linux, ubuntu-20.04core]
```
Comment (spencergilbert) [inline .github/workflows/regression.yml:321]:
```suggestion
    runs-on: [linux, ubuntu-20.04-4core]
```
?

Comment (neuronull) [reply .github/workflows/regression.yml]: good catch

**Review by jszwedko — COMMENTED** [review]
Could we add the details of each of these changes to the PR description? I'd like to record the "why" of these changes for posterity.

**Comment by neuronull** [comment]
> Could we add the details of each of these changes to the PR description? I'd like to record the "why" of these changes for posterity.

Updated, lmk if didn't hit the mark, can adjust.

**Review by jszwedko — APPROVED** [review]
Thanks for adding the context to the description!

**Pull request merged**